ksql-rest-app/src/test/resources/ksql-plan-schema/schema-final.json

```diff
@@ -0,0 +1,885 @@
+        },
+        "keyField" : {
+          "type" : "string"
+        },
+        "timestampColumn" : {
+          "$ref" : "#/definitions/TimestampColumn"
+        },
+        "kafkaTopicName" : {
+          "type" : "string"
+        },
+        "formats" : {
+          "$ref" : "#/definitions/Formats"
+        },
+        "windowInfo" : {
+          "$ref" : "#/definitions/WindowInfo"
+        }
+      },
+      "title" : "createTableV1",
+      "required" : [ "@type", "sourceName", "schema", "kafkaTopicName", "formats" ]
+    },
+    "RegisterTypeCommand" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "@type" : {
+          "type" : "string",
+          "enum" : [ "registerTypeV1" ],
+          "default" : "registerTypeV1"
+        },
+        "type" : {
+          "type" : "string"
+        },
+        "name" : {
+          "type" : "string"
+        }
+      },
+      "title" : "registerTypeV1",
+      "required" : [ "@type", "type", "name" ]
+    },
+    "DropSourceCommand" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "@type" : {
+          "type" : "string",
+          "enum" : [ "dropSourceV1" ],
+          "default" : "dropSourceV1"
+        },
+        "sourceName" : {
+          "type" : "string"
+        }
+      },
+      "title" : "dropSourceV1",
+      "required" : [ "@type", "sourceName" ]
+    },
+    "DropTypeCommand" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "@type" : {
+          "type" : "string",
+          "enum" : [ "dropTypeV1" ],
+          "default" : "dropTypeV1"
+        },
+        "typeName" : {
+          "type" : "string"
+        }
+      },
+      "title" : "dropTypeV1",
+      "required" : [ "@type", "typeName" ]
+    },
+    "QueryPlan" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "sources" : {
+          "type" : "array",
+          "items" : {
+            "type" : "string"
+          }
+        },
+        "sink" : {
+          "type" : "string"
+        },
+        "queryId" : {
+          "type" : "string"
+        },
+        "physicalPlan" : {
+          "$ref" : "#/definitions/ExecutionStep"
+        }
+      },
+      "required" : [ "sources", "sink", "queryId", "physicalPlan" ]
+    },
+    "StreamAggregate" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "@type" : {
+          "type" : "string",
+          "enum" : [ "streamAggregateV1" ],
+          "default" : "streamAggregateV1"
+        },
+        "properties" : {
+          "oneOf" : [ {
+            "$ref" : "#/definitions/DefaultExecutionStepProperties"
+          } ]
+        },
+        "source" : {
+          "$ref" : "#/definitions/ExecutionStep"
+        },
+        "formats" : {
+          "$ref" : "#/definitions/Formats"
+        },
```

Comment:
Can the `StreamAggregate` step change the format?  If not, then I don't think we need this.

Reply:
Yes, this is exactly what I had in mind here (and for groupby/join etc). When we write out to a changelog, or a repartition topic, we have a chance to use our own more optimal format. So it's beneficial to be able to specify the format for these internal operations.

Reply:
consider renaming to `internalFormat` then?